enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,335 @@
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+
+- Maturity levels
+    - [`alpha`, `beta`, `stable` in upstream Kubernetes][maturity-levels]
+    - `Dev Preview`, `Tech Preview`, `GA` in OpenShift
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, GA), regardless how the functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+#### Examples
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+1. The API will be unavailable from time to time during the installation.
+2. Coreos-installer cannot be used in the cloud environment.
+      
+## Alternatives
+
+### Installing using remote bootstrap node
+
+Run the bootstrap node in a HUB cluster as VM.
+This approach is appealing because it keeps the current installation flow.
+Requires external dependencies. 
+However, there are drawbacks:
+1. It will require Load balancer and DNS per installation.
+2. Deployments run remotely via L3 connection (high latency (up to 150ms), low BW in some cases), this isn't optimal for etcd cluster (one member is running on the bootstrap during the installation) 
+3. Running the bootstrap on the HUB cluster present a (resources) scale issue (~50*(8GB+4cores)), limits ACM capacity
+
+### Installing without liveISO
+
+Run the bootstrap flow on the node disk and clean up all the bootstrap residues once the node fully configured.
+This is very similar to the current enhancement installation approach but without the requirement to start from liveCD. 
+This approach advantage is that it will work on cloud environment. 
+The disadvantage is that it's more prune to result a single node deployment with bootstrap leftovers.
+
+
+### Installing using a baked Ignition file.
```

Comment:
We should document why we rejected this approach.

Reply:
https://github.com/eranco74/enhancements/pull/6